Logger.md

```diff
@@ -0,0 +1,107 @@
+# Logger
+
+`Logger` is an interface that helps decouple client classes from concrete implementations of a logging system.
+
+## Declaration
+
+```swift
+enum LogLevel {
+    case info
+    case warning
+    case error
+}
+
+protocol Logger {
+  func log(_ level: LogLevel, tag: String, message: String)
+  func print(tag: String, message: String)
+  func warning(tag: String, message: String) 
+  func error(tag: String, message: String)
+}
+```
+
+```kotlin
+// Kotlin
+TODO
+```
+
+## Usage
+
+```swift
+// Swift
+
+#if DEBUG
+  let logger = DeviceConsoleLogger()        
```

Comment:
Is this `DeviceConsoleLogger` a provided implementation as well? if so, we should document it. Are there other provided implementations?

Reply:
I've updated the sample code to highlight that these class names are examples. I could add them if there's consensus about this, but I would personally not add these classes to a library now. They are too simple and too likely to change from project to project. We might change our minds after seeing how different projects use this protocols and what concrete implementations they create.

Reply:
I would definitely include in the core a default DeviceConsoleLogger() class implementation that prints in the console (mostly doing a forward to print(....)). Why not otherwise?

Reply:
I included a concrete implementation: DeviceConsoleLogger.

Reply:
nice!

Reply:
Yes, the base implementation for the platform is required if not we will end up with multiples versions in every project and it's the same at the end.